src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -299,4 +299,22 @@ Copyright (c) .NET Foundation. All rights reserved.
     <IsNetCoreAppTargetingLatestTFM Condition="'$(IsNetCoreAppTargetingLatestTFM)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(_TargetFrameworkVersionWithoutV)' == '$(NETCoreAppMaximumVersion)'">true</IsNetCoreAppTargetingLatestTFM>
   </PropertyGroup>
 
+  <!--
+    Use Roslyn deployed with SDK for builds of SDK-style projects (regardless of whether the initiator is `dotnet` or `msbuild`).
+    See https://github.com/dotnet/sdk/blob/main/documentation/general/decouple-vs-and-net-sdk.md.
+  -->
+  <PropertyGroup>
+    <RoslynUseSdkCompiler Condition="'$(RoslynUseSdkCompiler)' == '' and
+      '$(RoslynUseMSBuildCompiler)' != 'true' and
+      '$(BuildWithNetFrameworkHostedCompiler)' != 'true' and
+      ('$(DOTNET_HOST_PATH)' != '' or '$(DOTNET_EXPERIMENTAL_HOST_PATH)' != '')">true</RoslynUseSdkCompiler>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(RoslynUseSdkCompiler)' == 'true'">
+    <RoslynTargetsPath Condition="'$(MSBuildRuntimeType)' == 'Core'">$(MSBuildThisFileDirectory)..\..\..\Roslyn</RoslynTargetsPath>
+    <RoslynTargetsPath Condition="'$(MSBuildRuntimeType)' != 'Core'">$(MSBuildThisFileDirectory)..\..\..\Roslyn\binfx</RoslynTargetsPath>
```

Comment:
Why do we need different targets for core and netfx? Today they don't differ so not sure why this is needed

Reply:
The one in `Roslyn` is `net9.0`, the one in `Roslyn\binfx` is `net472`. The first one is used from .net core msbuild, the latter one from .net framework msbuild. Otherwise it wouldn't work I think.